graph/flow/flow.go

```diff
@@ -0,0 +1,32 @@
+// Package flow provides control flow analysis functions.
+package flow
+
+import "gonum.org/v1/gonum/graph"
+
+// Graph is a control flow graph, a directed graph rooted at an entry node.
+type Graph interface {
+	// Entry returns the entry node of the control flow graph.
+	Entry() graph.Node
+	graph.Directed
+}
+
+// NewGraph returns a new control flow graph based on the given directed graph
+// and entry node ID.
+func NewGraph(g graph.Directed, entryid int64) Graph {
```

Comment:
s/entryid/rid/

Also, I'm not a fan of returning an interface type from a construction func and given the simplicity of this function, ISTM that just providing a type with the relevant exposed fields would do the job.

Reply:
Fair point, that would also work.

Reply:
One question that arise. Do we even need `flow.Graph` at all? The reason I added it was so that functions taking `g graph,Directed, root int64` would instead take `g flow.Graph`. However, with your suggestions, we'd still use the first approach for the API, and then we probably won't have a reason to keep `flow.Graph`.

Reply:
Yes, I did think about whether it should go away, but keeping them bundled once you have them is nice. You have come to the same conclusion.